simple-weather/simple-weather/Controller/ViewController.swift

```diff
@@ -29,28 +30,43 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
         
         if CLLocationManager.locationServicesEnabled() {
             locationManager.delegate = self
-            locationManager.desiredAccuracy = kCLLocationAccuracyBest
+            locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers
             locationManager.startUpdatingLocation()
         }
         
-        locationLabel.text = url
-        
         let dateFormatter = DateFormatter()
         dateFormatter.dateFormat = "MMMM dd, yyyy";
         let myDate = dateFormatter.string(from: Date.init())
         dateLabel.text = "Today, \(myDate)"
     }
     
+    
+    
     func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
         guard let locValue: CLLocationCoordinate2D = manager.location?.coordinate else { return }
-        print("locations = \(locValue.latitude) \(locValue.longitude)")
-        let jsonUrl = "http://api.openweathermap.org/data/2.5/weather?lat=\(locValue.latitude)&lon=\(locValue.longitude)&appid=4ec61b9764720fc34bc6123d2169ab81"
-        print(jsonUrl)
         
-        //fetchCurrentWeather
+        APIService().getData(coordinates: locValue) { currentWeather in
+            DispatchQueue.main.async {
+                 self.locationLabel.text = currentWeather.cityName
+                self.tempLabel.text = "\(String(Int((currentWeather.currentTemp - 273.15))))°"
+                self.weatherLabel.text = currentWeather.weatherType
+                
+                self.weatherImage.image = UIImage(named: "Clear")
+                
+                if currentWeather.weatherType == "Clouds" {
```

Comment:
Redundant if-else if checks. `weatherType` is `String` no need to check "Coulds" == "Clouds",  "Rain" == "Rain" etc. All could be written with one line `self.weatherImage.image = UIImage(named: currentWeather.weatherType)`